code/game/machinery/recharger.dm

```diff
@@ -34,142 +34,153 @@
+		return
+	. += span_notice("- \The [charging] is not reporting a power level.")
 
-/obj/machinery/recharger/proc/setCharging(new_charging)
-	charging = new_charging
-	if (new_charging)
+/obj/machinery/recharger/Entered(atom/movable/arrived, atom/old_loc, list/atom/old_locs)
+	if(is_type_in_typecache(arrived, allowed_devices))
+		charging = arrived
 		START_PROCESSING(SSmachines, src)
 		update_use_power(ACTIVE_POWER_USE)
 		finished_recharging = FALSE
 		using_power = TRUE
 		update_appearance()
-	else
+	return ..()
+
+/obj/machinery/recharger/Exited(atom/movable/gone, direction)
+	if(gone == charging)
+		if(!QDELETED(charging))
+			charging.update_appearance()
+		charging = null
 		update_use_power(IDLE_POWER_USE)
 		using_power = FALSE
 		update_appearance()
+	return ..()
 
-/obj/machinery/recharger/attackby(obj/item/G, mob/user, params)
-	if(G.tool_behaviour == TOOL_WRENCH)
-		if(charging)
-			to_chat(user, span_notice("Remove the charging item first!"))
-			return
-		set_anchored(!anchored)
-		power_change()
-		to_chat(user, span_notice("You [anchored ? "attached" : "detached"] [src]."))
-		G.play_tool_sound(src)
-		return
-
-	var/allowed = is_type_in_typecache(G, allowed_devices)
-
-	if(allowed)
-		if(anchored)
-			if(charging || panel_open)
-				return 1
-
-			//Checks to make sure he's not in space doing it, and that the area got proper power.
-			var/area/a = get_area(src)
-			if(!isarea(a) || a.power_equip == 0)
-				to_chat(user, span_notice("[src] blinks red as you try to insert [G]."))
-				return 1
-
-			if (istype(G, /obj/item/gun/energy))
-				var/obj/item/gun/energy/E = G
-				if(!E.can_charge)
-					to_chat(user, span_notice("Your gun has no external power connector."))
-					return 1
-
-			if(!user.transferItemToLoc(G, src))
-				return 1
-			setCharging(G)
-
-		else
-			to_chat(user, span_notice("[src] isn't connected to anything!"))
-		return 1
-
-	if(anchored && !charging)
-		if(default_deconstruction_screwdriver(user, "recharger", "recharger", G))
-			update_appearance()
-			return
+/obj/machinery/recharger/attackby(obj/item/attacking_item, mob/user, params)
+	if(!is_type_in_typecache(attacking_item, allowed_devices))
+		return ..()
+
+	. = TRUE
+	if(!anchored)
+		to_chat(user, span_notice("[src] isn't connected to anything!"))
+		return
+	if(charging || panel_open)
+		return
+
+	var/area/our_area = get_area(src) //Check to make sure user's not in space doing it, and that the area got proper power.
+	if(!isarea(our_area) || our_area.power_equip == 0)
+		to_chat(user, span_notice("[src] blinks red as you try to insert [attacking_item]."))
+		return
 
-		if(panel_open && G.tool_behaviour == TOOL_CROWBAR)
-			default_deconstruction_crowbar(G)
+	if (istype(attacking_item, /obj/item/gun/energy))
+		var/obj/item/gun/energy/energy_gun = attacking_item
+		if(!energy_gun.can_charge)
+			to_chat(user, span_notice("Your gun has no external power connector."))
 			return
+	user.transferItemToLoc(attacking_item, src)
```

Comment:
```suggestion
	user.transferItemToLoc(attacking_item, src)
	return TRUE
```

yeah it's best to have that explicitness right here so people don't get confounded by it for whatever weird debugger reason we have in the standards